cpp/include/raft/core/device_resources_snmg.hpp

```diff
@@ -16,226 +16,99 @@
-#include <nccl.h>
-
-#include <memory>
-#include <numeric>
 #include <vector>
 
 namespace raft {
 
 /**
- * @brief SNMG (single-node multi-GPU) resource container object that stores a NCCL clique and all
- * necessary resources used for calling device functions, cuda kernels, libraries and/or NCCL
- * communications on each GPU. Note the `device_resources_snmg` object can also be used as a classic
+ * @brief SNMG (single-node multi-GPU) resource container object that tracks resources for each GPU.
+ * Note the `device_resources_snmg` object can also be used as a classic
  * `device_resources` object. The associated resources will be the ones of the GPU used during
  * object instantiation and a GPU switch operation will be ordered during the retrieval of said
  * resources.
  *
  * The `device_resources_snmg` class is intended to be used in a single process to manage several
- * GPUs. Please note that NCCL communications are the responsibility of the user. Blocking NCCL
- * calls will sometimes require the use of several threads to avoid hangs.
+ * GPUs.
  */
 class device_resources_snmg : public device_resources {
  public:
   /**
-   * @brief Construct a SNMG resources instance with all available GPUs
+   * @brief Construct a SNMG base resources instance with all available GPUs
    */
   device_resources_snmg() : device_resources()
   {
     RAFT_CUDA_TRY(cudaGetDevice(&main_gpu_id_));
-
-    // set root rank
     int root_rank = 0;
-    raft::resource::set_nccl_clique_root_rank(*this, root_rank);
-
-    // initialize the NCCL clique
-    std::vector<raft::resources>& clique = raft::resource::get_nccl_clique(*this);
-    _init_clique(clique);
-    _init_nccl_comms(clique);
+    raft::resource::set_root_rank(*this, root_rank);
 
+    // initialize all resources
+    std::vector<raft::resources>& world_resources = raft::resource::get_multi_gpu_resource(*this);
+    _init_world(world_resources);
     RAFT_CUDA_TRY(cudaSetDevice(main_gpu_id_));
   }
 
   /**
-   * @brief Construct a SNMG resources instance with a subset of available GPUs
+   * @brief Construct a SNMG base resources instance with a subset of available GPUs
    *
-   * @param[in] device_ids List of device IDs to be used by the NCCL clique
+   * @param[in] device_ids List of device IDs to be used
    */
   device_resources_snmg(const std::vector<int>& device_ids) : device_resources()
   {
     RAFT_CUDA_TRY(cudaGetDevice(&main_gpu_id_));
-
-    // set root rank
     int root_rank = 0;
-    raft::resource::set_nccl_clique_root_rank(*this, root_rank);
-
-    // initialize the NCCL clique
-    std::vector<raft::resources>& clique = raft::resource::get_nccl_clique(*this);
-    _init_clique(clique, device_ids);
-    _init_nccl_comms(clique);
+    raft::resource::set_root_rank(*this, root_rank);
 
-    RAFT_CUDA_TRY(cudaSetDevice(main_gpu_id_));
+    // initialize resources for the given device ids
+    std::vector<raft::resources>& clique = raft::resource::get_multi_gpu_resource(*this);
```

Comment:
Maybe instead of clique we should call it something like "world_resources"? NCCL creates a clique, but we do not when we're just using "cudaSetDevice" manually.